openedx/core/lib/xblock_runtime/runtime.py

```diff
@@ -0,0 +1,108 @@
+from lxml import etree
+
+from xblock.core import XBlock, XBlockAside
+from xblock.field_data import ReadOnlyFieldData, SplitFieldData
+from xblock.fields import Scope, ScopeIds
+from xblock.runtime import Runtime, IdReader, IdGenerator, NullI18nService, MemoryIdManager
+
+from openedx.core.lib.xblock_utils import xblock_local_resource_url
+from xmodule.modulestore.inheritance import inheriting_field_data
+from .blockstore_kvs import collect_parsed_fields
+
+
+class XBlockRuntime(Runtime):
```

Comment:
How feasible is it to move this out of process in the medium to long term?

Reply:
When I first started writing this I tried to write it as a new runtime and avoid importing anything else in edx-platform. However, at some point I stopped doing that and just started importing things from `xmodule.blah` when I needed them. I will try to cut back on that again.

So far, other than the code to get definition_keys from usage_ids for split mongo blocks, I haven't had to add any code which couldn't be separated. But I suspect there's going to be more of that when I get to wiring up more of the runtime services and the pseudo-services (the runtime attributes like `replace_urls`, anonymous student IDs, xqueue, etc.)